nestjs/src/courses/course-mentors/course-mentors.service.ts

```diff
@@ -0,0 +1,191 @@
+      .andWhere("c.type <> 'interview'")
+      .getManyAndCount();
+
+    return [courseTasks.map(task => task.id), count];
+  }
+
+  public async getMentorsWithStats(courseId: number): Promise<MentorDetails[]> {
+    const [courseTasksIds, count] = await this.getMentorCheckerCourseTaskIds(courseId);
+
+    const [records, checkedCountByMentor, lastCheckedDateByMentor] = await Promise.all([
+      this.getMentorsRecordsWithDetails(courseId),
+      this.getCheckedTasksCount(courseId, courseTasksIds),
+      this.getLastCheckedDates(courseId, courseTasksIds),
+    ]);
+
+    return this.constructMentorsWithStats(records, checkedCountByMentor, lastCheckedDateByMentor, count);
+  }
+
+  public async searchMentors(
+    courseId: number,
+    searchText: string,
+  ): Promise<{ id: number; githubId: string; name: string }[]> {
+    const entities = await this.mentorsRepository
+      .createQueryBuilder('mentor')
+      .innerJoin('mentor.user', 'user')
+      .addSelect(['user.id', 'user.firstName', 'user.lastName', 'user.githubId'])
+      .where('"mentor"."courseId" = :courseId', { courseId })
+      .andWhere('"mentor"."isExpelled" = false')
+      .andWhere(
+        `
+        (
+          "user"."githubId" ILIKE :searchText OR
+          "user"."firstName" ILIKE :searchText OR
+          "user"."lastName" ILIKE :searchText OR
+          CONCAT("user"."firstName", ' ', "user"."lastName") ILIKE :searchText
+        )
+      `,
+        { courseId, searchText },
+      )
+      .limit(20)
+      .getMany();
+
+    const result = entities.map(entity => ({
+      id: entity.id,
+      githubId: entity.user.githubId,
+      name: PersonDto.getName(entity.user),
+    }));
+
+    return result;
+  }
+
+  private async getMentorsRecordsWithDetails(courseId: number) {
+    return await this.mentorsRepository
+      .createQueryBuilder('mentor')
+      .innerJoin('mentor.user', 'user')
+      .addSelect(UsersService.getPrimaryUserFields())
+      .addSelect(['user.contactsEpamEmail'])
+      .leftJoin('mentor.students', 's')
+      .leftJoin('mentor.stageInterviews', 'si')
+      .leftJoin('mentor.taskChecker', 'tc')
+      .leftJoin('mentor.interviewResults', 'ir')
+      .leftJoin('tc.courseTask', 't', 't.type = :type', { type: 'interview' })
+      .addSelect(['s.id', 's.isExpelled', 's.isFailed'])
+      .addSelect(['si.id', 'si.isCompleted'])
+      .addSelect(['tc.id', 'tc.courseTaskId'])
+      .addSelect(['t.id', 't.type'])
+      .addSelect(['ir.id'])
+      .where(`"mentor"."courseId" = :courseId`, { courseId })
+      .orderBy('mentor.createdDate')
+      .getMany();
+  }
+
+  private buildTaskResultSubQuery(courseTasksIds: number[]) {
+    const subQuery = this.taskResultRepository.createQueryBuilder('t').select('t.lastCheckerId', 'lastCheckerId');
+
+    if (courseTasksIds.length) {
+      subQuery.where('t.courseTaskId IN (:...ids)', { ids: courseTasksIds });
+    } else {
+      // to prevent empty array
+      subQuery.where('1 = 0');
```

Comment:
could u please clarify how does this help?  ie this is always falsy

Reply:
The old solution was to concat `0` into `courseTaskIds`, the current solution `1 = 0` was suggested by Valery.
I believe this helps us to have a `null` instead empty array in the result.

Reply:
@aaliakseyenka I double-checked and concluded that this comment is misleading. This part is not for preventing an empty array, but preserving it :)

Reply:
right the comment is misleading. IMO prev solution was more reliable without branches. But up to u to decide. at least plz fix a comment

Reply:
> right the comment is misleading. IMO prev solution was more reliable without branches. But up to u to decide. at least plz fix a comment

The comment has already been fixed, but not pushed yet. I like your reasoning about branches

Reply:
Fixed